File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        int len = progresses.length;
        
        
        int idx = 0;
        int count = 0;
        while(idx != len){
            count = 0;
            for(int i=idx; i<len; i++){
                progresses[i] += speeds[i];
            }
            
            for(int i=idx; i<len; i++){
                if(progresses[i] >= 100){
                    idx = i+1;
                    count += 1;
                }else{
                    break;
                }
            }
            
            if(count != 0) answer.add(count);
        }
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    n = len(progresses)
    idx = 0
    
    while idx < n:
        count = 0
        diff = 100 - progresses[idx]
        days = math.ceil(diff / speeds[idx])
        
        for i in range(idx, n):
            progresses[i] += (days * speeds[i])
        
        for i in range(idx, n):
            if progresses[i] >= 100:
                idx += 1
                count += 1
            else :
                break;
        answer.append(count)
        
    return answer